src/Http/Routing/src/HostAttribute.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    /// <summary>
+    /// Attribute for providing host metdata that is used during routing.
+    /// </summary>
+    [DebuggerDisplay("{DebuggerToString(),nq}")]
+    public class HostAttribute : IHostMetadata
```

Comment:
Sealed, and needs to be an attribute 😆

Reply:
Details, details. It's an attribute, it says so in the name. [It's the vibe of the thing.](https://www.youtube.com/watch?v=nMuh33BMZYY)